### ISSUE TITLE
Should +Y correspond to left in PyTorch3D's coordinate system?

### ISSUE THREAD

**Issue by AndreiBarsan** [facebookresearch/pytorch3d#625]
I have been reading the PyTorch3D transforms documentation and I have one question about the geometric conventions.

Specifically, there is one example that I am still confused about: on the [Wikipedia Rotation matrix article](https://en.wikipedia.org/wiki/Rotation_matrix#Basic_rotations), towards the bottom of the "Basic Rotations" section, there is an example of a positive 90° yaw rotation, about the Z axis.

The example from Wikipedia is right-handed, with positive rotation angles corresponding to CCW rotations. As far as I understand, this matches the [conventions from PyTorch3D](https://pytorch3d.readthedocs.io/en/latest/modules/transforms.html#pytorch3d.transforms.Transform3d).

In the example, applying this rotation to a vector pointing along the X axis, i.e., (`[1, 0, 0]`) rotates it so it aligns with Y, i.e., (`[0, 1, 0]`). One can easily visualize how this transform would work---looking down at, say, a table, a vector pointing forward is simply rotated by 90° to point left. And since Y = left, this corresponds to `[0, 1, 0]`.

However, a similar transform (which I may, however, have a mistake in) yields `[0, -1, 0]` in PyTorch3D.

This would indicate that the rotation about the Z axis corresponds to a rotation of the coordinate frame itself, and not a rotation of the points upon which the `Transform3d` acts. However, if one transforms the above point using `Transform3d().translate(10, 10, 10)`, then one gets `[11, 10, 10]`, indicating that the translation part translates the point itself, and not the coordinate system, since translating the coordinate system by that amount would make the point's coordinates negative.

My question is therefore: Is this expected behavior on the part of the `Transform3d` object? 

## Steps to reproduce this behavior (rotation part)

```python
from pytorch3d.transforms import Transform3d, euler_angles_to_matrix
import torch
import math
yaw90 = euler_angles_to_matrix(torch.tensor([0.0, 0.0, math.pi / 2]), "XYZ")
unit_x_vec = torch.tensor([1.0, 0.0, 0.0])
transform = Transform3d().rotate(yaw90)
transform.transform_points(unit_x_vec.view(1, 3))
```

Interactive session results:
```python
>>> from pytorch3d.transforms import Transform3d, euler_angles_to_matrix
>>> import torch
>>> import math
>>> yaw90 = euler_angles_to_matrix(torch.tensor([0.0, 0.0, math.pi / 2]), "XYZ")
>>> yaw90
tensor([[-4.3711e-08, -1.0000e+00,  0.0000e+00],
        [ 1.0000e+00, -4.3711e-08,  0.0000e+00],
        [ 0.0000e+00,  0.0000e+00,  1.0000e+00]])
>>> unit_x_vec = torch.tensor([1.0, 0.0, 0.0])
>>> transform = Transform3d().rotate(yaw90)
>>> transform
<pytorch3d.transforms.transform3d.Transform3d object at 0x7f45a2f15760>
>>> transform.get_matrix()
tensor([[[-4.3711e-08, -1.0000e+00,  0.0000e+00,  0.0000e+00],
         [ 1.0000e+00, -4.3711e-08,  0.0000e+00,  0.0000e+00],
         [ 0.0000e+00,  0.0000e+00,  1.0000e+00,  0.0000e+00],
         [ 0.0000e+00,  0.0000e+00,  0.0000e+00,  1.0000e+00]]])
>>> transform.transform_points(unit_x_vec.view(1, 3))
tensor([[-4.3711e-08, -1.0000e+00,  0.0000e+00]])
```

**Comment by nikhilaravi** [comment]
@AndreiBarsan If I understand correctly you were trying to apply a 90 degree CCW rotation about the Z axis to a vector pointing along the +X axis? 

From the wikipedia article:

> each of these basic vector rotations appears counterclockwise when **the axis about which they occur points toward the observer**, the coordinate system is right-handed, and the angle θ is positive. 

In PyTorch3D the coordinate system is right handed, however the +X axis points left, the +Y is up and +Z into the screen (i.e. away from the observer).  In that case a +90 CCW about the Z axis of a vector in the direction `[1, 0, 0]` would give `[0, -1, 0]` which is what you observed.

**Comment by AndreiBarsan** [comment]
This makes perfect sense, thank you for clarifying @nikhilaravi! My brain had "overfit" to the forward-left-up conventions from ROS. 😅 

Do you think people would find it helpful if I sent over a PR clarifying this convention in `transform3d.py`?

**Comment by nikhilaravi** [comment]
Sure you can add that to `transforms3d.py`. We explain the coordinate system in [this doc](https://github.com/facebookresearch/pytorch3d/blob/master/docs/notes/assets/world_camera_image.jpg) but it could be useful to have it documented in transforms as well.

**Comment by DanielTakeshi** [comment]
@nikhilaravi Sorry for bumping an old issue but I have a similar question as @AndreiBarsan and am wondering about the terminology. A right handed coordinate system can indeed be used with +X pointing left (thumb), +Y up (index), and +Z into screen / away from me (middle finger). 

However, when we talk about a "positive rotation about an axis" does that refer to a rotation about the *positive* z axis? If so, and if positive rotation means counter-clockwise (as these docs suggest https://pytorch3d.readthedocs.io/en/latest/modules/transforms.html) then isn't a 90 degree counterclockwise rotation supposed to transform (1,0,0) into (0,1,0) instead of (0,-1,0)? The code does seem to produce (0,-1,0) so I'm trying to better understand why this is the case.

In particular, from this text here:

> each of these basic vector rotations appears counterclockwise when the axis about which they occur points toward the observer, the coordinate system is right-handed, and the angle θ is positive.

It seems to suggest that, if the angle of the z coordinate is *pointing towards the observer* (i.e., from screen *to* me) then a positive angle (e.g., 90 degrees) is counterclockwise. This aligns with my understanding of how rotations work. But here, z is pointing *into* the screen, hence a positive rotation is *clockwise* and that seems like it would take (1,0,0) into (0,1,0)? Here are some drawings to clarify my thinking.

To the left, this is what PyTorch3d seems to be returning. But to the right, this is how I thought things would work. I assumed that: if z is pointing to me, then positive rotation about its axis is anti-clockwise, then that will move the vector along the xy plane in the anti-clockwise direction?

![Screenshot from 2022-05-07 20-49-58](https://user-images.githubusercontent.com/5532096/167277044-ddcdb181-2bae-4b7a-b1dc-905c361c82e4.png)

Am I missing something in my logic?
